libraries/AP_Compass/AP_Compass_HMC5843.cpp

```diff
@@ -31,7 +31,11 @@
 
 #include <AP_Math/AP_Math.h>
 #include <AP_HAL/AP_HAL.h>
+#if  CONFIG_HAL_BOARD != HAL_BOARD_REVOMINI
 #include <AP_HAL/utility/sparse-endian.h>
+#else
+#include <stm32f4xx.h>
+#endif
```

Comment:
humn... no.. too many ifdefs. Let's implement the missing bits in the missing directory? Then this common code is clean as it was before.